glommio/src/io/dma_file_stream.rs

```diff
@@ -519,25 +493,20 @@ impl DmaStreamReader {
     }
 
     /// Allows access to the buffer that holds the current position with no
-    /// extra copy
-    /// In order to use this API, one must guarantee that reading the specified
-    /// length may not cross into a different buffer.  Users of this API are
-    /// expected to be aware of their buffer size (selectable in the
-    /// [`DmaStreamReaderBuilder`]) and act accordingly.
+    /// extra copy.
+    ///
+    /// This function returns a tuple of a [`ReadResult`] and a [`usize`]. The
```

Comment:
I am a fan of the approach. Well done.

Not sure I am a fan of the unnamed tuple. Why do you think this is preferable to a specialized struct ?

Reply:
Right, I agree. 

Come to think of it, this API should work like POSIX `read,` i.e., it should return what was read instead of what is left to read. What is left to read is actually very hard to know, given we could hit EOF at any time.

If I return what was read instead, then I don't need to return a tuple because, by definition, you can already get that with `ReadResult::len().` That will simplify things even more. Nice!

Reply:
Done!